File: day1/part1/main.py
depths = []
inc_count = 0

with open("day1/input.txt", "r") as inputf:
  for line in inputf:
    depths.append(int(line.strip()))
  
for i in range(1999):
  if depths[i+1] > depths[i]:
    inc_count += 1

print(f"Depths larger than the previous: {inc_count}")

File: day1/part1/main.c
#include <stdio.h>
#include <stdlib.h>

#define INPUT_LEN 2000

int main(void)
{
  FILE *inputf;
  int depths[INPUT_LEN];
  int inc_count = 0;
  int line = 0;

  inputf = fopen("./day1/input.txt", "r");

  if (inputf == NULL)
  {
    printf("Error while opening file");
    exit(1);
  }

  while (!feof(inputf))
  {
    fscanf(inputf, "%d", &depths[line]);
    line++;
  }

  fclose(inputf);

  for (int i = 0; i < INPUT_LEN - 1; i++)
  {
    if (depths[i + 1] > depths[i])
    {
      inc_count++;
    }
  }

  printf("larger than previous: %d\n", inc_count);

  return 0;
}
